lib/redmine/menu_manager/top_menu_helper.rb

```diff
@@ -132,9 +132,12 @@ def render_login_partial
   end
 
   def render_module_top_menu_node(items = more_top_menu_items)
-    render_drop_down_menu_node link_to(l(:label_modules), '#', title: l(:label_modules), class: 'icon5 icon-version'),
-                               items,
-                               id: 'more-menu'
+    return '' if User.current.anonymous? and Setting.login_required?
+    render_drop_down_menu_node(
+      link_to(l(:label_modules), '#', title: l(:label_modules), class: 'icon5 icon-version'),
+      items,
+      id: 'more-menu'
+    )
```

Comment:
Why are you not applying that logic to the menu items configured in `config/initializers/menus.rb` directly? That way, it would be more granular.